build-aux/snap/snapcraft.yaml

```diff
@@ -32,6 +32,15 @@ parts:
     build-snaps: [go/1.13/stable]
     override-pull: |
       snapcraftctl pull
+      # set version, this needs dpkg-parsechangelog (from dpkg-dev) and git
+      sudo -E apt-get install -y dpkg-dev git
+      if sh -x ./mkversion.sh --output-only | grep "dirty"; then
+          echo "dirty git tree during build detected:"
+          git status
+          git diff
+          exit 1
+      fi
```

Comment:
thanks for this, I had a similar patch but which didn't fail the build and instead put the diff into the snap to look at, but I think this is a better idea to just fail the build

Reply:
hmm although actually when testing my build-packages suggestion I realized why I didn't do that in my local patch, since that means to build the snapd snap at all you have to commit all changes and you cannot have any unstaged changes, which is rather unfortunate for local development on snapd... I will propose my work-around of including the diff into the built snapd snap instead of failing the build and we can discuss there I suppose

Reply:
My first PR of the year, and only PR today because I am supposed to have today off: https://github.com/snapcore/snapd/pull/11197